### PR TITLE
fix unstable test in sink hang

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix case-3 in https://github.com/pingcap/ticdc/issues/838

The transactions are sent to two sink workers, we must ensure the sink worker with `MySQLSinkHangLongTime` injected is executed before the sink worker with `MySQLSinkExecDMLError` injected returns.

### What is changed and how it works?

Add a delay in sink worker before injected `MySQLSinkExecDMLError` returns.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -716,6 +716,9 @@ func (s *mysqlSink) prepareDMLs(rows []*model.RowChangedEvent, replicaID uint64,
 
 func (s *mysqlSink) execDMLs(ctx context.Context, rows []*model.RowChangedEvent, replicaID uint64, bucket int) error {
 	failpoint.Inject("MySQLSinkExecDMLError", func() {
+		// Add a delay to ensure the sink worker with `MySQLSinkHangLongTime`
+		// failpoint injected is executed first.
+		time.Sleep(time.Second * 2)
```
Comment (overvenus) [inline cdc/sink/mysql.go:721]: What about use the failpoint expr "1*2s -> return(true)"?

Comment (amyangfei) [reply cdc/sink/mysql.go:721]: `1*sleep(2000)->1*return(true)`
should be triggered two times, but the first time DMLs will be executed successfully

**Pull request merged**